site/docs/plugins/entity-parser.md

```diff
@@ -0,0 +1,237 @@
+- Setting the value to `false` will skip sanitization, keeping the output text as the original.
+  This is not recommended, as it may result in incorrect rendering and make your application vulnerable to XSS attacks.
+  Ensure proper handling if you choose this option.
+- If you provide a function, it will be used instead of the default sanitizer.
+
+```ts
+const myTextSanitizer: TextSanitizer = (options: TextSanitizerOption): string =>
+  // Replace dangerous character
+  options.text.replaceAll(/[&<>"']/, (match) => {
+    switch (match) {
+      case "&":
+        return "&amp;";
+      case "<":
+        return "&lt;";
+      case ">":
+        return "&gt;";
+      case '"':
+        return "&quot;";
+      case "'":
+        return "&#x27;";
+      default:
+        return match;
+    }
+  });
+
+// Implement the sanitizer.
+const entitiesParser = new EntitiesParser({ textSanitizer: myTextSanitizer });
+```
+
+## Cases When It's Better Not to Use This Package
+
+If you face problems similar to those listed below, you might be able to resolve them without using this package.
+
+### Copying and Forwarding the Same Message
+
+Use [`forwardMessage`](https://core.telegram.org/bots/api#forwardmessage) to forward messages of any kind.
+
+You can also use the [`copyMessage`](https://core.telegram.org/bots/api#copymessage) API, which performs the same action but does not include a link to the original message.
+[`copyMessage`](https://core.telegram.org/bots/api#copymessage) behaves like copying the message and sending it back to Telegram, making it appear as a regular message rather than a forwarded one.
+
+```ts
+bot.on(":text", async (ctx) => {
+  // The target chat id to send.
+  const chatId = -946659600;
+  // Forward the current message without a link to the original message.
+  await ctx.copyMessage(chatId);
+  // Forward the current message with a link to the original message.
+  await ctx.forwardMessage(chatId);
+});
+```
+
+### Replying to Messages with Modified Text Format
+
+You can easily reply to incoming messages using HTML, Markdown, or entities.
+
+```ts
+bot.on(":text", async (ctx) => {
+  // Reply using HTML
+  await ctx.reply("<b>bold</b> <i>italic</i>", { parse_mode: "HTML" });
+  // Reply using Telegram Markdown V2
+  await ctx.reply("*bold* _italic_", { parse_mode: "MarkdownV2" });
+  // Reply with entities
+  await ctx.reply("bold italic", {
+    entities: [
+      { offset: 0, length: 5, type: "bold" },
+      { offset: 5, length: 6, type: "italic" },
+    ],
+  });
+});
+```
+
+::: tip Use parse-mode for a Better Formatting Experience
+
+Use the official [`parse-mode`](./parse-mode) plugin for a better experience constructing formatted messages.
+:::
+
+## Plugin Summary
+
+- Name: entity-parser
+- Package: <https://jsr.io/@qz/telegram-entities-parser>
+- Source: <https://github.com/quadratz/telegram-entities-parser>
```

Comment:
Why we are not using default construction?

```ts
## Plugin Summary

- Name: string
- [Source](url)
- [Reference](/ref/ref/)
```

Should we stay `- Package: <url>` or change to `- [Package](url)`

Reply:
This is correct. It only sanitizes the input text, not the result itself. For instance, if the entities are
```json
  "text": "Bold & Italic <script>",
  "entities": [
   {
    "offset": 0,
    "length": 4,
    "type": "bold"
   },
   {
    "offset": 7,
    "length": 6,
    "type": "italic"
   }
  ]
```

The original output will be:
```html
<b>Bold</b> & <i>Italic<i/> <script>
```

Then it will be sanitized into:
```html
<b>Bold</b> &amp; <i>Italic<i/> &lt;script&gt;
```

Reply:
Also, the name of the plugins is usually put in back quotes.

Reply:
Good catch. I will make a separate PQ for that.

Reply:
Or should I push the changes to this PQ and sync them to all languages myself since it does not require translation?
```md
- Name: `entity-parser`
- [Package](https://jsr.io/@qz/telegram-entities-parser)
- [Source](https://github.com/quadratz/telegram-entities-parser)
```

Reply:
Yeah feel free to do taht

Reply:
Fixed in https://github.com/grammyjs/website/pull/1121/commits/4c41245006594d9daacdbe38a81b7177acb8c552 and https://github.com/grammyjs/website/pull/1121/commits/d7377d1673fa06fc847052c8921400f7de59281c